src/callbacks/wandb_callbacks.py

```diff
@@ -196,3 +196,40 @@ def on_validation_epoch_end(self, trainer, pl_module):
 
             self.preds.clear()
             self.targets.clear()
+            
+
+class ImagePredictionLogger(Callback):
+    """Logs a validation batch and their predictions to wandb.
+       Example adapted from:
+       https://wandb.ai/wandb/wandb-lightning/reports/Image-Classification-using-PyTorch-Lightning--VmlldzoyODk1NzY
+    """
+
+    def __init__(self, num_samples):
+        super().__init__()
+        self.num_samples = num_samples
+
+    def on_sanity_check_start(self, trainer, pl_module):
+        self.ready = False
+
+    def on_sanity_check_end(self, trainer, pl_module):
+        """Start executing this callback only after all validation sanity checks end."""
+        self.ready = True
+
+    def on_validation_epoch_end(self, trainer, pl_module):
+        # get a validation batch from the calidation dat loader
+        val_samples = next(iter(trainer.datamodule.val_dataloader()))
+        val_imgs, val_labels = val_samples[0], val_samples[1]
+
+        # run the batch through the network
+        val_imgs = val_imgs.to(device=pl_module.device)
+        logits = pl_module(val_imgs)
+        preds = torch.argmax(logits, axis=-1)
+
+        # log the images as wandb Image
+        trainer.logger.experiment[0].log({
+            'Images':[wandb.Image(x, caption=f'Pred:{pred}, Label:{y}')
```

Comment:
You could use:
```suggestion
            f"Images/{trainer.logger.experiment[0].name}":[wandb.Image(x, caption=f"Pred:{pred}, Label:{y}")
```
This way we're using `/` character wchich will create separate section in wandb dashboard called "Images"

Reply:
Seems like good idea, however when I  test it the captions (Pred and Label) are no longer added to the images. 
Do you know why?

Reply:
The captions are visible for me when I enter into run view 
![image](https://user-images.githubusercontent.com/43458894/115972767-a3dbf480-a550-11eb-9b1e-6a2036487a23.png)
The previous version looks the same for me in run view:
![image](https://user-images.githubusercontent.com/43458894/115972786-c0782c80-a550-11eb-8225-7093fef15e4b.png)
The difference is only in multi experiment dashboard view:
![image](https://user-images.githubusercontent.com/43458894/115972803-e271af00-a550-11eb-9e8b-273ed78582f2.png)
But this view always shows only one image from each experiment so it's not very useful either way...
I hope wandb will fix this some day.

Reply:
Ok I checked again on my run view and the captions appear and disappear with the size of the box where the images are placed in.
Should we add a comment somewhere that the box needs to be dragged bigger to see the captions?

Reply:
Nah, I would just let it be. Those image views in wandb behave weirdly and I assume they will just fix it over time.